src/main/java/com/gmail/goosius/siegewar/events/PreSiegeWarStartEvent.java

```diff
@@ -0,0 +1,76 @@
+package com.gmail.goosius.siegewar.events;
+
+import com.palmergames.bukkit.towny.object.Nation;
+import com.palmergames.bukkit.towny.object.Town;
+import com.palmergames.bukkit.towny.object.TownBlock;
+import org.bukkit.block.Block;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+import org.jetbrains.annotations.NotNull;
+
+public class PreSiegeWarStartEvent extends Event implements Cancellable {
+
+    private static final HandlerList handlers = new HandlerList();
+    private final Town townOfAttackingPlayer;
+    private final Nation nationOfAttackingPlayer;
+    private final Block block;
+    private final TownBlock townBlock;
+    private final Town defendingTown;
+    private boolean isCancelled;
+    private String cancellationMsg;
```

Comment:
The cancellationMsg should have a default value, something like "Siege prevented by another plugin."